pvm.py

```diff
@@ -35,116 +43,230 @@ def _init_logger():
+			return
+
+		# Log command and execute time.
+		_logger.info("Command: %s, execute time: %s.", command, next_time.time())
+		
+		# Wait until current time is equal to next_time
+		while True:
+			now = datetime.now()
+			now = now.replace(microsecond=0)
+			if now.time() >= next_time.time():
+				break
+			sleep(0.005)
 	
     # Get value
 	if len(args) == 6:
 		value = args[5]
 
+	# Catch failed parse command
 	try:
 		# File command
 		if command=="file":
-			_logger.info("File set: %s", PREFIX_PATH + value)
-			IS_FILE_SET = True
-			media = OMXPlayer(PREFIX_PATH + value, dbus_name='org.mpris.MediaPlayer2.omxplayer', args=['--loop'])
-			media.pause()
-			VIDEO_PATH = value
+			# define displayNum here
+			if "|" not in value:
+				displayNum = 1
+			else:
+				displayNum = 2
+
+			if displayNum == 1:
+				VIDEO_PATH_ONE = PEFIX_PATH + value
+				_logger.info("File set: %s with 1 video output.", VIDEO_PATH_ONE)
+				if IS_FILE_SET:
+					_logger.info("The file has already been set, please hit stop first.")
+					return
+				# Change flags
+				IS_FILE_SET = True
+				canStart = True
+				canPause = False
+				media1 = OMXPlayer(VIDEO_PATH_ONE, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=['--loop'])
+				media1.pause()
+			else:
+				VIDEO_PATH_ONE, VIDEO_PATH_TWO = parse_value(value)
+				VIDEO_PATH_ONE = PEFIX_PATH + VIDEO_PATH_ONE
+				VIDEO_PATH_TWO = PEFIX_PATH + VIDEO_PATH_TWO
+				_logger.info("File set: %s, %s with 2 video outputs.", VIDEO_PATH_ONE, VIDEO_PATH_TWO)
+				if IS_FILE_SET:
+					_logger.info("The files have already been set, please hit stop first.")
+					return
+				# Change flags
+				IS_FILE_SET = True
+				canStart = True
+				canPause = False
+				media1 = OMXPlayer(VIDEO_PATH_ONE, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=['--loop', '--display', '2'])
+				media1.pause()
+				media2 = OMXPlayer(VIDEO_PATH_TWO, dbus_name='org.mpris.MediaPlayer2.omxplayer2', args=['--loop', '--display', '7'])
+				media2.pause()
 			return
 
 		# If file is unset, then we should not execute any command below.
 		if not IS_FILE_SET:
 			_logger.info("Command %s failed because of the file is unset.", command)
 			return
 
-		if command=="start":
-			if media.can_play():
-				media.play()
+		# If Raspberry Pi has two displays.
+		if displayNum == 2:
+			if command=="start":
+				if canStart:
+						# Get position for media1 and media2
+						pos1 = media1.position()
+						pos2 = media2.position()
+						# Always set position for the fast media to slow one.
+						if pos1 > pos2:
+							media1.set_position(pos2)
+						elif pos1 < pos2:
+							media2.set_position(pos1)		
+						pos1 = media1.position()
+						_logger.info("media 1&2 position: %s", pos1)
+						# Play both media
+						media1.play()
+						media2.play()
+						# Change canPause and canStart
+						canPause = True
+						canStart = False
+						_logger.info("%s command success.", command)
```

Comment:
```suggestion
					# Get position for media1 and media2
					pos1 = media1.position()
					pos2 = media2.position()
					# Always set position for the fast media to slow one.
					if pos1 > pos2:
						media1.set_position(pos2)
					elif pos1 < pos2:
						media2.set_position(pos1)		
					pos1 = media1.position()
					_logger.info("media 1&2 position: %s", pos1)
					# Play both media
					media1.play()
					media2.play()
					# Change canPause and canStart
					canPause = True
					canStart = False
					_logger.info("%s command success.", command)
```

this is going to be problematic.... it implies that you always need to start both displays at the same time. Displays are not independent in this way!